architecture/spmc.rst

```diff
@@ -0,0 +1,322 @@
+====
+SPMC
+====
+
+This document describes the SPMC(S-EL1) implementation for OP-TEE.
+More information on the SPMC can be found in the FF-A specification can be
+found in the
+`FF-A spec <https://developer.arm.com/documentation/den0077/latest>`_.
+
+.. toctree::
+ :numbered:
+
+SPMC Responsibilities
+=====================
+
+The SPMC is a critical component in the FF-A flow. Some of it mayor
+responsibilities are:
+
+- Initialisation and run-time management of the SPs:
+        The SPMC component is responsible for initialisation of the
+        Secure Partitions (Loading the Image, Setting up the stack, HEAP, ...).
+- Routing messages between endpoints:
+        The SPMC is responsible for passing FF-A messages from The Normal World
+        to SPs and back. It also needs to to pass FF-A messages between SPs
+- Memory management:
+        The SPMC is responsible for the memory management of the SPs. Memory
+        can be shared between SPs and from a SP to the Normal World.
+
+This document describes OP-TEE as a S-EL1 SPMC.
+
+Secure Partitions
+=================
+Secure Partitions (SPs) are the endpoints used in the FF-A protocol. When
+OP-TEE is used as a SPMC SPs run primarily inside S-EL0.
+
+The SPMC also exposes the OP-TEE kernel as an secure endpoint itself. This is
+used to handle all OP-TEE legacy operations. All legacy operations will be
```

Comment:
It might be a bit early to call the communication with TAs legacy operations. I'm not sure if SPs will ever be able to replace TAs.